spec/requests/subject_and_classses_spec.rb

```diff
@@ -0,0 +1,173 @@
+      headers['Authorization'] = "Bearer #{@jwt_token}"
+    end
+
+    it 'returns a token' do
+      expect(response.headers['Authorization']).to be_present
+    end
+
+    describe 'Admin can assign subjects to class' do
+      before do
+        assign_subjects(false)
+      end
+
+      it 'returns 200' do
+        expect(response).to have_http_status(200)
+      end
+
+      it 'should return a success message' do
+        expect(response.body).to include('Successfully updated the class.')
+      end
+
+      it 'should return the assigned subjects' do
+        expect(JSON.parse(response.body)['data'].size).to eq(2)
+      end
+
+      it 'should return the assigned subjects and class details and its type' do
+        data = [{
+          'id' => SubAndClass.first.id.to_s,
+          'type' => 'subject_and_class',
+          'attributes' =>
+                  {
+                    'id' => SubAndClass.first.id,
+                    'created_at' => SubAndClass.first.created_at.as_json,
+                    'subject' => { 'id' => Subject.first.id, 'name' => Subject.first.name.to_s,
+                                   'created_at' => Subject.first.created_at.as_json },
+                    'teaching_class' => { 'id' => TeachingClass.first.id, 'name' => TeachingClass.first.name.to_s,
+                                          'created_at' => TeachingClass.first.created_at.as_json }
+                  }
+        },
+                {
+                  'id' => SubAndClass.last.id.to_s,
+                  'type' => 'subject_and_class',
+                  'attributes' =>
+                  {
+                    'id' => SubAndClass.last.id,
+                    'created_at' => SubAndClass.last.created_at.as_json,
+                    'subject' => { 'id' => Subject.last.id, 'name' => Subject.last.name.to_s,
+                                   'created_at' => Subject.last.created_at.as_json },
+                    'teaching_class' => { 'id' => TeachingClass.last.id, 'name' => TeachingClass.last.name.to_s,
+                                          'created_at' => TeachingClass.last.created_at.as_json }
+                  }
+                }]
+        expect(JSON.parse(response.body)['data']).to eq(data)
+      end
+    end
+
+    describe 'When admin assigns already assigned subjects to a class' do
+      before do
+        assign_subjects(true)
+      end
+
+      it 'returns status 400' do
+        expect(response).to have_http_status(400)
+      end
+
+      it 'should return a error message' do
+        expect(response.body).to include('Subject has already been taken')
+      end
+    end
+
+    describe 'Admin can remove subjects from the class' do
+      before do
+        remove_subjects
+      end
+
+      it 'returns 200' do
+        expect(response).to have_http_status(200)
+      end
+
+      it 'should return a success message' do
+        expect(response.body).to include('Successfully removed subjects from the class.')
```

Comment:
@josephelias-rubyians  Can you also check if the records are removed also from the teaching_class.subjects perspective?